cloud/ec2.go

```diff
@@ -39,10 +39,10 @@ type EC2ProviderSettings struct {
 	AMI string `mapstructure:"ami" json:"ami,omitempty" bson:"ami,omitempty"`
 
 	// If set, overrides key from credentials
-	AWSKeyID string `mapstructure:"aws_access_key_id" bson:"aws_access_key_id,omitempty"`
+	AWSKeyID string `mapstructure:"aws_access_key_id" json:"aws_access_key_id,omitempty" bson:"aws_access_key_id,omitempty"`
 
 	// If set, overrides secret from credentials
-	AWSSecret string `mapstructure:"aws_secret_access_key" bson:"aws_secret_access_key,omitempty"`
+	AWSSecret string `mapstructure:"aws_secret_access_key" json:"aws_access_key_id,omitempty" bson:"aws_secret_access_key,omitempty"`
```

Comment:
Is there a reason why these didn't have JSON keys? Wondering if it's a secrecy thing. Otherwise unmarshalling/marshalling birch Documents with BSON doesn't work after re-reading and writing from an API route. (I think all setting documents will actually need json and bson tags.)

Reply:
Yeah, I suspect it's that, though I think that with the API, this is not an issue, since there's an intermediate type. I suspect someone was being extra cautious. I'm not 100% sure that this isn't returned in an API route anywhere, but I can't think of where. I checked the host page, and it's not in the source.

Reply:
@john-m-liu do you have any additional context? Just want to be careful.

Reply:
It looks like for host BuildFromService we specifically pull out the distro information we want. Do we need to specifically nil this out for the distro BuildFromService? But then I would think Update() would also erase the key/secret.